dbt/adapters/impala/connections.py

```diff
@@ -60,14 +59,35 @@ class ImpalaCredentials(Credentials):
         'user':'username'
     }
 
+    @classmethod
+    def __pre_deserialize__(cls, data):
+        data = super().__pre_deserialize__(data)
+        if 'database' not in data:
+            data['database'] = None
+        return data
+
+    def __post_init__(self):
+        # impala classifies database and schema as the same thing
+        if (
+            self.database is not None and
+            self.database != self.schema
+        ):
+            raise dbt.exceptions.RuntimeException(
+                f'    schema: {self.schema} \n'
+                f'    database: {self.database} \n'
+                f'On Impala, database must be omitted or have the same value as'
+                f' schema.'
+            )
```

Comment:
Do we need a conditional check to throw exception if profile file is missing schema.

Reply:
No, schema is a required parameter as opposed to database being optional and is already handled in the standard checks in the dbt flow.